Framework/PythonInterface/plugins/algorithms/MaskShadowedPixels.py

```diff
@@ -0,0 +1,178 @@
+        self.declareProperty(mantid.api.FileProperty('OutputFilename', '', action=mantid.api.FileAction.Save, extensions=['.gss']),
+                             doc='Name of the GSAS file to save to')
+
+        # TODO - Add missing input properties
+
+    def validateInputs(self):
+        """Validate input properties
+        (virtual method)
+        """
+        issues = dict()
+
+        return issues
+
+    def PyExec(self):
+        """
+        Main method to execute the algorithm
+        """
+        # Prepare input workspace: normalize by solid angle
+        raw_workspace = self.getProperty('InputWorkspace').value
+        assert raw_workspace, 'Input workspace cannot be None'
+        intensity_array = self.prepare_input(raw_workspace)  # return (101378,) array
+
+        # Parse YMAL configuration
+        config = self.parse_yaml(self.getProperty('ConfigurationFile').value)
+        assert isinstance(config, dict)
+
+        # Get instrument information
+        instrument_info = self.set_nomad_constants()
+
+        # Initialize the empty mask list
+        mask_array = np.zeros(shape=(instrument_info.num_pixels,), dtype='float')
+        assert mask_array
+
+        # Determine thresholds for a given tube (part 1)
+        # TASK 335
+        self.determine_tube_threshold(intensity_array, instrument_info, config)
+        # END OF Part1 ---
+
+        # Calculate the median for a particular Group of banks (part 2)
+        # TASK 331
+        self.calculate_banks_medians(config)
+        # END OF Part2 ----
+
+        # Determine thresholds for a given bank (part 3)
+        # TASK 333
+        self.determine_banks_thredsholds(config)
+        # END OF Part3 ---
+
+        # Process for output
+
+        return
+
+    def prepare_input(self, workspace):
+        """Calculate count on each detector pixels and normalized by solid angle correction
+
+        Parameters
+        ----------
+        workspace
+
+        Returns
+        -------
+        numpy.ndarray
+            shape = (101378,)
+
+        """
+        # Calculate the count on each detector pixel
+        # Get counts array: shape = (101378, Bins)
+        vec_intensity = workspace.extractY()
+        # Sum along Y axis (axis = 1): shape = (101378, )
+        vec_intensity = vec_intensity.sum(axis=1)
+
+        # Calculate solid angle
+        solid_angle_array = self.calculate_solid_angle(workspace).extractY().flatten()
+
+        vec_intensity /= solid_angle_array
+
+        return vec_intensity
+
+    @staticmethod
+    def calculate_solid_angle(workspace):
+        """Calculate solid angle for each detector
+
+        Parameters
+        ----------
+
+
+        Returns
+        -------
+        numpy.ndarray
+            shape = (N, ) or (N, 1), where N is the number of detectors
+
+        """
+        assert workspace
+        # TODO - To be implemented
+        return np.ndarray()
```

Comment:
another empty function that does nothing.
If it is no longer needed, we should remove it to avoid confusion.